csharp/src/Ice/ConnectionFactory.cs

```diff
@@ -1581,33 +1504,25 @@ private void SetState(int state)
             {
                 case StateActive:
                     {
-                        if (_state != StateHolding) // Can only switch from holding to active.
+                        if (_state != StateInactive) // Can only switch from inactive to active.
                         {
                             return;
                         }
                         if (_acceptor != null)
                         {
                             if (_communicator.TraceLevels.Network >= 1)
                             {
-                                var s = new StringBuilder("accepting ");
-                                s.Append(_endpoint.Transport());
-                                s.Append(" connections at ");
-                                s.Append(_acceptor.ToString());
-                                _communicator.Logger.Trace(_communicator.TraceLevels.NetworkCat, s.ToString());
+                                _communicator.Logger.Trace(_communicator.TraceLevels.NetworkCat,
+                                    $"accepting {_endpoint.Transport()} connections at {_acceptor}");
                             }
                             _adapter!.ThreadPool.Register(this, SocketOperation.Read);
                         }
-
-                        foreach (Connection connection in _connections)
-                        {
-                            connection.Activate();
-                        }
                         break;
                     }
 
-                case StateHolding:
+                case StateInactive:
                     {
-                        if (_state != StateActive) // Can only switch from active to holding.
+                        if (_state != StateActive) // Can only switch from active to inactive.
```

Comment:
This is not correct. Presumably the factory's state can only progress from Inactive to Active to Closed to Finished.
And we actually call this private SetState only 3 times, and never with the StateInactive parameter - so this code is never reached.

Reply:
I fixed this.